pages/employee/index.vue

```diff
@@ -0,0 +1,142 @@
+const allowNumericValue = (event: InputEvent) => {
+    const input = event.target as HTMLInputElement;
+    input.value = input.value.replace(/\D/g, '');
+    formChanged.value = true;
+};
+const validate = (employee: any): FormError[] => {
+    const errors = []
+    if (!employee.first_name) errors.push({ path: 'first_name', message: 'First Name Required' })
+    if (!employee.last_name) errors.push({ path: 'last_name', message: 'Last Name Required' })
+
+    if (!employee.employe_type) errors.push({ path: 'employe_type', message: 'Employee Type Required' })
+    if (!employee.mobile) {
+        errors.push({ path: 'mobile', message: 'Mobile Number Required' })
+    } else if (!/^\d{10}$/.test(employee.mobile)) {
+        errors.push({ path: 'mobile', message: 'Mobile Number should be  10 digits ' });
+    }
+    if (!employee.pincode) {
+        errors.push({ path: 'pincode', message: 'Pincode Required' })
+    } else if (!/^\d{6}$/.test(employee.pincode)) {
+        errors.push({ path: 'pincode', message: 'Pincode should be  6 digits ' });
+    }
+    formChanged.value = true;
+
+    return errors
+}
+
+const incomeTitle = computed(() => {
+    return employee.value.employe_type === 'Business' ? 'Income' : 'Salary';
+});
+function submitEmployee() {
+    if (pincodeError.value === '') {
+        sessionStorage.setItem('employee', JSON.stringify(employee.value));
+        formChanged.value = false;
+    }
+}
+
+async function onError(event: FormErrorEvent) {
+    const element = document.getElementById(event.errors[0].id)
+    element?.focus()
+    element?.scrollIntoView({ behavior: 'smooth', block: 'center' })
+}
+async function checkPincode() {
+    try {
+        const response = await fetch(`https://api.postalpincode.in/pincode/${employee.value.pincode}`)
+        const data = await response.json()
+        if (data && data.length > 0) {
+            const pincodeDetails = data[0]
+            if (pincodeDetails.Status === 'Error') {
+                pincodeError.value = 'Invalid pincode. Please enter a valid pincode.'
+            } else {
+                pincodeError.value = ''
+            }
+        } else {
+            pincodeError.value = 'Invalid pincode. Please enter a valid pincode.'
+        }
+    } catch (error) {
+        console.error('Error checking pincode:', error)
+        pincodeError.value = 'An error occurred while checking the pincode. Please try again later.'
+    }
+
+}
+onBeforeRouteLeave((to, from, next) => {
+    if (formChanged.value) {
+        const confirmed = window.confirm(
+            "Changes in the form. Do you want to save them before leaving?"
+        );
+        if (confirmed) {
+            submitEmployee()
+            next();
+        } else {
+            next(false); // Prevent leaving the route
+        }
+    } else {
+        next();
+    }
+})
+onMounted(() => {
+    if (sessionStorage && sessionStorage.getItem('employee') && typeof sessionStorage.getItem('employee') === 'string') {
```

Comment:
Is it mandatory to perform a type check for the 'employee' sessionStorage item?

Reply:
Resolved